errors/api_error.go

```diff
@@ -6,11 +6,14 @@ import (
 )
 
 type ApiError struct {
+	// ErrorCode is different from the HTTP Status Code as it represents a Nullstone-specific error code
+	ErrorCode int
+
 	Err error
 }
 
-func NewApiError(err error) ApiError {
-	return ApiError{Err: err}
+func NewApiError(errorCode int, err error) ApiError {
```

Comment:
Does this mean that we have to define an error code for every error we return through the api? This could be a good thing, just making sure I understand first.

Reply:
Correct, for unknown errors, use the default 0.